### PR TITLE
Copy Jobs to batch/v1 group

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by erictune** [comment]
@lavalamp this should be working well enough for you to take a look at it.  I'm waiting to see what integration and e2e testing says, but unit tests pass.

File: hack/verify-api-reference-docs.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ TMP_ROOT="${KUBE_ROOT}/_tmp"
 
 echo "diffing ${API_REFERENCE_DOCS_ROOT} against freshly generated docs"
 ret=0
-diff -NauprBZ -I 'Last update' --exclude=*.md "${API_REFERENCE_DOCS_ROOT}" "${OUTPUT_DIR}" || ret=$?
+diff -NauprB -I 'Last update' --exclude=*.md "${API_REFERENCE_DOCS_ROOT}" "${OUTPUT_DIR}" || ret=$?
```
Comment (lavalamp) [inline hack/verify-api-reference-docs.sh]: Is this diff intentional? seems unrelated?

File: pkg/api/testapi/testapi.go [hunk 2]
```diff
@@ -78,6 +81,12 @@ func init() {
 			internalGroupVersion: extensions.SchemeGroupVersion,
 		}
 	}
+	if _, ok := Groups[batch.GroupName]; !ok {
+		Groups[batch.GroupName] = TestGroup{
+			externalGroupVersion: unversioned.GroupVersion{Group: batch.GroupName, Version: registered.GroupOrDie(batch.GroupName).GroupVersion.Version},
+			internalGroupVersion: extensions.SchemeGroupVersion,
```
Comment (lavalamp) [inline pkg/api/testapi/testapi.go]: I missed this in @piosz's change-- can you move the following `if _, ok := Groups[extensions.GroupName]; !ok {` section above the batch & autoscaling setup code? otherwise, referencing the extensions group here may not do what we want.

**Comment by lavalamp** [comment]
No major comments.

File: hack/verify-api-reference-docs.sh (hunk 1 above)
Comment (erictune) [inline hack/verify-api-reference-docs.sh]: I can't regenerate on my machine without this.  But I can take it out of this PR.

File: pkg/api/testapi/testapi.go (hunk 2 above)
Comment (erictune) [inline pkg/api/testapi/testapi.go]: ok

**Comment by erictune** [comment]
Both review comments addressed.  Also, commented out a scaler test which was failing: I will remove Scaler from Job in a subsequent PR, as already requested by bgrant0607

**Comment by erictune** [comment]
@k8s-bot please unit test again, flake #21451

**Comment by erictune** [comment]
@lavalamp review comments addressed and passing all tests except for a flake on the last run.  Ready for LGTM.

**Comment by erictune** [comment]
Found problem in kubect, pushed fix.

**Comment by soltysh** [comment]
LGTM

**Comment by erictune** [comment]
I intend the commit named "Comment out test to fix later" to merge.

**Comment by erictune** [comment]
Applying LGTM that @lavalamp gave over chat.

**Comment by erictune** [comment]
@k8s-bot e2e test this issue #21463

**Comment by erictune** [comment]
@k8s-bot e2e test this issue #21463

**Comment by lavalamp** [comment]
\o/
